hawkbit/lib_operation_status.go

```diff
@@ -24,7 +24,7 @@ type OperationStatus struct {
 	// Software is required for a remove or cancelRemove operation, absent in case of install/download/cancel.
 	Software []*DependencyDescription `json:"software,omitempty"`
 	// Progress represents the progress indicator in percentage.
-	Progress int `json:"progress,omitempty"`
+	Progress *int `json:"progress,omitempty"`
```

Comment:
Is using a pointer really needed? A new OperationStatus instance seems to be created for every status change. It worsens the readability of the logs, as the progress is not clear from it's address:
```
2024/10/25 11:12:58.401915  [Hawkbit] DEBUG   Set status/lastOperation property to: &{other_correlation_id DOWNLOADING 0xc000060000 [] 0xc000584000  }
2024/10/25 11:12:58.648696  [Hawkbit] DEBUG   Set status/lastOperation property to: &{other_correlation_id DOWNLOADING 0xc000060180 [] 0xc000584040  }
2024/10/25 11:12:58.813242  [Hawkbit] DEBUG   Set status/lastOperation property to: &{other_correlation_id DOWNLOADING 0xc000060400 [] 0xc000584080  }
2024/10/25 11:12:59.149099  [Hawkbit] DEBUG   Set status/lastOperation property to: &{other_correlation_id DOWNLOADING 0xc000594000 [] 0xc0001ba0f0  }
2024/10/25 11:12:59.444500  [Hawkbit] DEBUG   Set status/lastOperation property to: &{other_correlation_id DOWNLOADING 0xc000518000 [] 0xc000520000  }
2024/10/25 11:12:59.656123  [Hawkbit] DEBUG   Set status/lastOperation property to: &{other_correlation_id DOWNLOADING 0xc000594100 [] 0xc0001ba1a0  }
2024/10/25 11:12:59.905828  [Hawkbit] DEBUG   Set status/lastOperation property to: &{other_correlation_id DOWNLOADING 0xc000594200 [] 0xc0001ba1f0  }
2024/10/25 11:13:00.118258  [Hawkbit] DEBUG   Set status/lastOperation property to: &{other_correlation_id DOWNLOADING 0xc000594300 [] 0xc0001ba280  }
```

Reply:
Logging concern is now addressed with enhancing logging of last operation status. 
In the previous version of logging, SoftwareModuleID and SoftwareDependencies were also logged with only pointer address. Now a map is created to place all the last operation status with the keys and values and then same is logged into a file.

Reply:
2024/11/03 01:36:32.518928  [Hawkbit] DEBUG   Set status/lastOperation property to: map[correlationId:other_correlation_id softwareModule:map[name:kanto-deb version:1.0.0] status:STARTED]
2024/11/03 01:36:32.519468  [Hawkbit] DEBUG   Set status/lastOperation property to: map[correlationId:other_correlation_id progress:0 softwareModule:map[name:kanto-deb version:1.0.0] status:DOWNLOADING]
2024/11/03 01:36:36.095805  [Hawkbit] DEBUG   Set status/lastOperation property to: map[correlationId:other_correlation_id progress:1 softwareModule:map[name:kanto-deb version:1.0.0] status:DOWNLOADING]
2024/11/03 01:36:36.355329  [Hawkbit] DEBUG   Set status/lastOperation property to: map[correlationId:other_correlation_id progress:2 softwareModule:map[name:kanto-deb version:1.0.0] status:DOWNLOADING]
2024/11/03 01:36:36.948011  [Hawkbit] DEBUG   Set status/lastOperation property to: map[correlationId:other_correlation_id progress:3 softwareModule:map[name:kanto-deb version:1.0.0] status:DOWNLOADING]
2024/11/03 01:36:37.273580  [Hawkbit] DEBUG   Set status/lastOperation property to: map[correlationId:other_correlation_id progress:4 softwareModule:map[name:kanto-deb version:1.0.0] status:DOWNLOADING]
2024/11/03 01:36:37.562039  [Hawkbit] DEBUG   Set status/lastOperation property to: map[correlationId:other_correlation_id progress:5 softwareModule:map[name:kanto-deb version:1.0.0] status:DOWNLOADING]
2024/11/03 01:36:37.821551  [Hawkbit] DEBUG   Set status/lastOperation property to: map[correlationId:other_correlation_id progress:6 softwareModule:map[name:kanto-deb version:1.0.0] status:DOWNLOADING]
2024/11/03 01:36:38.063579  [Hawkbit] DEBUG   Set status/lastOperation property to: map[correlationId:other_correlation_id progress:7 softwareModule:map[name:kanto-deb version:1.0.0] status:DOWNLOADING]
2024/11/03 01:36:38.325928  [Hawkbit] DEBUG   Set status/lastOperation property to: map[correlationId:other_correlation_id progress:8 softwareModule:map[name:kanto-deb version:1.0.0] status:DOWNLOADING]
2024/11/03 01:36:38.597579  [Hawkbit] DEBUG   Set status/lastOperation property to: map[correlationId:other_correlation_id progress:9 softwareModule:map[name:kanto-deb version:1.0.0] status:DOWNLOADING]